tests/phpunit/tests/wp-token-map/wpTokenMap.php

```diff
@@ -0,0 +1,404 @@
+			'article',
+			$map->read_token( $text, 0, $skip_bytes ),
+			"Should have matched 'a' but matched '" . substr( $text, 0, $skip_bytes ) . "' instead."
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[].
+	 */
+	public static function data_short_substring_matches_of_each_other() {
+		$map = WP_Token_Map::from_array(
+			array(
+				'a'       => 'article',
+				'aa'      => 'defensive weapon',
+				'ar'      => 'country code',
+				'arizona' => 'state name',
+			)
+		);
+
+		return array(
+			'single character'    => array( $map, 'antarctica is a continent', 'a' ),
+			'duplicate character' => array( $map, 'aaaaahhhh, he exclaimed', 'aa' ),
+			'different character' => array( $map, 'argentina is a country', 'ar' ),
+			'full word'           => array( $map, 'arizona was full of copper', 'arizona' ),
+		);
+	}
+
+	/**
+	 * Ensures that Token Map searches at appropriate starting offset.
+	 *
+	 * @ticket 60698.
+	 *
+	 * @dataProvider data_html5_test_dataset
+	 *
+	 * @param string $token       Token to find.
+	 * @param string $replacement Replacement string for token.
+	 */
+	public function test_reads_token_at_given_offset( $token, $replacement ) {
+		$document = "& another {$token} & then some";
+		$map      = self::get_html5_token_map();
+
+		$skip_bytes = 0;
+		$this->assertFalse(
+			$map->read_token( $document, 0, $skip_bytes ),
+			"Shouldn't have found token at start of document."
+		);
+
+		$response = $map->read_token( $document, 10, $skip_bytes );
+
+		$this->assertSame(
+			strlen( $token ),
+			$skip_bytes,
+			"Found the wrong length for token '{$token}'."
+		);
+
+		$this->assertSame(
+			$response,
+			$replacement,
+			'Found the wrong replacement value for the token.'
+		);
+	}
+
+	/**
+	 * Ensures that all given tokens exist inside a constructed Token Map.
+	 *
+	 * @ticket 60698.
+	 *
+	 * @dataProvider data_html5_test_dataset
+	 *
+	 * @param string $token       Token to find.
+	 * @param string $replacement Not used in this test.
+	 */
+	public function test_detects_all_tokens( $token, $replacement ) {
+		$map = self::get_html5_token_map();
+
+		$this->assertTrue(
+			$map->contains( $token ),
+			"Should have found '{$token}' inside the Token Map, but didn't."
+		);
+
+		$double_escaped_token = str_replace( '&', '&amp;', $token );
+		$this->assertFalse(
+			$map->contains( $double_escaped_token ),
+			"Should not have found '{$double_escaped_token}' in Token Map, but did."
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[].
+	 */
+	public function data_html5_test_dataset() {
+		$html5 = self::get_test_input_array( 'HTML5' );
+		$cases = array();
+
+		$this->assertSame(
+			self::KNOWN_COUNT_OF_ALL_HTML5_NAMED_CHARACTER_REFERENCES,
+			count( $html5 ),
+			'Found the wrong number of HTML5 named character references: confirm the entities.json file."'
+		);
+
+		foreach ( $html5 as $token => $replacement ) {
+			$cases[ $token ] = array( $token, $replacement );
+		}
+
+		return $cases;
```

Comment:
This seems like an ideal place to yield cases, we don't need to collect and return them all at the same time:

```suggestion
		foreach ( $html5 as $token => $replacement ) {
			yield $token => array( $token, $replacement );
		}
```

The `$cases = array();` line above could also be removed.

Reply:
Great suggestion. Yielding in 2cf613f866c1321983f5f4385c0334585a43d75c